numpy/core/src/multiarray/mapping.c

```diff
@@ -2567,6 +2578,31 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     NPY_END_THREADS;
     Py_DECREF(intp_type);
     return 0;
+
+indexing_error:
+
+    if (mit->size == 0) {
+        PyObject *err_type = NULL, *err_value = NULL, *err_traceback = NULL;
+        PyErr_Fetch(&err_type, &err_value, &err_traceback);
+        /* 2020-04-02, NumPy 1.19 */
+        if (DEPRECATE(
+                "Out of bound index found. This was previously ignored "
+                "when the indexing result contained no elements. "
+                "In the future the index error will be raised. This error "
+                "occurs either due to an empty slice, or if an array has zero "
+                "elements even before indexing.\n"
+                "(Use `warnings.simplefilter('error')` to raise the warning "
+                "and get more details on the invalid index.)") < 0) {
+            npy_PyErr_ChainExceptions(err_type, err_value, err_traceback);
+            return -1;
```

Comment:
> Use `warnings.simplefilter('error')` to raise the warning

Which warning does "the warning" refer to, this DeprecationWarning or some other?

Reply:
expanded the "the" to "this DeprecationWarning" and squashed the commits.

Reply:
So now I am confused. This message **is** the `DeprecationWarning`. It seems redundant, or am I missing something?

Also, your rebase caught a change in a submodule ...

Reply:
Ahh, maybe you meant "to turn this warning into an error and get more details ..."

Reply:
ah, dang, yes and thanks for noticing the submodule change, `commit -a` is dangerous right now :)

Reply:
Thanks, yes, adopted... I could just delete the whole thing as well.